### PR TITLE
CSA-78: Add change password, modal and linter settings

### PR DESCRIPTION
**Pull request by VovaLyapota**

### CODE REVIEW

File: src/components/EditProfileForm/EditProfileForm.tsx
Comment (tymnik) [inline src/components/EditProfileForm/EditProfileForm.tsx:1]: Fix the order of imports (libraries-components-styles)

File: src/components/ChangePasswordForm/ChangePasswordForm.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import { selectIsLoading, selectToken } from '../../store/user/selectors';
+import { changePassword } from '../../store/user/thunks';
+import Input from '../Input';
+import Spinner from '../Spinner';
+import styles from './ChangePasswordForm.module.css';
+
+const ChangePasswordForm = ({ closeModal }: { closeModal: () => void }) => {
+  const [shouldShowOldPassword, setShouldShowOldPassword] = useState(false);
+  const [shouldShowNewPassword, setShouldShowNewPassword] = useState(false);
+  const { id } = useAppSelector(selectToken);
```
Comment (StanislavKharkiv) [inline src/components/ChangePasswordForm/ChangePasswordForm.tsx]: I think variables with boolean value better start with is `isShowOldPassword`
It's just recommendation not rule)

File: src/components/EditProfileForm/EditProfileForm.tsx [hunk 2]
```diff
@@ -20,8 +20,12 @@ import Icon from '../Icon';
 import Input from '../Input';
 import Spinner from '../Spinner';
 import styles from './EditProfileForm.module.css';
+import { useState } from 'react';
+import { Modal } from '../Modal';
+import ChangePasswordForm from '../ChangePasswordForm';
 
 const EditProfileForm = () => {
+  const [shouldShowModal, setShouldShowModal] = useState(false);
   const isLoading = useAppSelector(selectIsLoading);
```
Comment (StanislavKharkiv) [inline src/components/EditProfileForm/EditProfileForm.tsx]: The same, we can write `isShowModal`

File: src/components/Modal/Modal.module.css [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  height: 100vh;
+  background-color: var(--modal-backdrop);
+
+  backdrop-filter: blur(10px);
+}
+
+.modal {
+  position: relative;
+  width: 440px;
+  height: fit-content;
```
Comment (StanislavKharkiv) [inline src/components/Modal/Modal.module.css:18]: this width is ok for mobile device? Probably better max-width and min-width ?

File: src/components/Modal/Modal.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        (event instanceof MouseEvent &&
+          event.target instanceof HTMLDivElement &&
+          event.target.id === 'modal-overlay')
+      ) {
+        closeModal();
+      }
+    };
+
+    if (isOpen) {
+      document.body.style.overflow = 'hidden';
```
Comment (StanislavKharkiv) [inline src/components/Modal/Modal.tsx]: I think better put handleClose out of useEffect

please remove       `document.body.style.overflow = 'hidden';`

File: src/components/Modal/Modal.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        createPortal(
+          <div id="modal-overlay" className={styles.overlay}>
+            <div className={clsx(styles.modal, className)}>
+              {shouldShowButton && (
+                <Button
+                  variant="icon"
+                  className={styles.button}
+                  onClick={closeModal}
+                  icon={<Icon id="trash" />}
+                />
```
Comment (StanislavKharkiv) [inline src/components/Modal/Modal.tsx]: Why icon trash? Must be X

File: src/api/user.ts [hunk 6]
```diff
@@ -1,14 +1,66 @@
...
+  id: UserId;
+  refresh: string;
+}) => {
+  const { access } = await getAccessToken(refresh);
+  setToken(access);
+
+  const user = await getUserById(id);
+
+  return { user, token: { refresh, id } };
+};
```
Comment (StanislavKharkiv) [inline src/api/user.ts:62]: I meant write this in thunk. We have different api functions with use inside asyncThunk.

**Review by StanislavKharkiv — APPROVED** [review]

**Pull request merged**